Part 2 (of 2) of changes based on Daniel Migault's review (#200)
ietf-teep/architecture: draft-ietf-teep-architecture.md

1. Build two apps:          |       |                |            |
                            |       |                |            |
   (a) Untrusted            |       |                |            |

>        App - 2a. Supply --->|--------- 3. Get App -->|            |

Author: ```suggestion
       App - 2a. Supply --> | --- 3. Install ------> |            |
```

1. Build two apps:          |       |                |            |
                            |       |                |            |
   (a) Untrusted            |       |                |            |
       App - 2a. Supply --->|<---------- 3. Get App -|            |

>                             |----------------------->|            |

Reviewer: --> means the source or destination? We use it as destination: A-->B means that the object will be installed in B. To this, I think we should also change "3. Get App to use -->", and 4 below "-- 4. Get TA -->".

Author: The arrow directions confused Daniel who expected them to show who initiated the step.  Since Get App is initiated from the device not the TAM, the old 3 Install was confusing in his review.  Hence the change to show it initiates from the device in this example, and the TAM returns the app in 694.  That was my attempt to address his comment.

Reviewer 2: If it means who initiates, "Get an app cert" should use the other direction: an app developer goes to CA to get a cert where developer initiates the certificate enrollment and requests. Similarly, "Get a TAM cert" should be that TAM initiates enrollment to a CA. I am fine either convention but feel that the "Get App" and "Get Cert" are using different conventions. The original "3. Install" is consistent with "2. supply". Can we clarify with Daniel on this without change? Or we change the arrows for the "cert" arrows?

Reviewer 2: How about explicitly make the second return arrow as a step?

Still I feel that this diagram used two conventions about the arrow between certificate enroll and app download. Hope use the same. I can go with either one. Changing the cert arrows will do. How do you think?

Author: Good point, I have changed the arrows back and instead added a parenthetical phrase to clarify.  How's it look now?

Reviewer: I like the explanation in the text. Sorry for one more comment. Do we have to change "install app" to "get app"? "Get app to" is what this means. In the protocol spec, we do use API "TrustedAppInstall"; we don't use "get app". TAM sends app to TEE, not a TEE to go out to get an app. The original word "install app" sounds right, in particular, in coherence with the arrow. Your parenthetical phrase resolve Daniel's comment. Can this work? I am OK if you strongly like to use "Get App".

>                             |----------------------->|            |

Reviewer 2: ```suggestion
                            |--- 3.2 Send the app -->|            |
```

                            |       |                |            |
   (b) TA -- 2b. Supply ----------->|<-- 4. Get TA --|            |

>                             |       |--------------->|            |

Reviewer 2: ```suggestion
                            |       |- 4.2 Send TA ->|            |
```

                            |       |                |            |
~~~~
{: #experience title="Example Developer Experience"}

{{experience}} shows an example where the same developer builds and signs
two applications: 1) an Untrusted Application; 2) a TA
that provides some security functions to be run inside
a TEE.  This example assumes that the developer, the TEE, and the TAM have

> previously been provisioned with certificatates.  

Reviewer: certificatates --> certificates

> previously been provisioned with certificatates.  

Author: ```suggestion
previously been provisioned with certificates.  
```

   (a) Untrusted            |       |                |            |
       App - 2a. Supply --> | --- 3. Install ------> |            |
                            |       |                |            |

>    (b) TA -- 2b. Supply ----------->|- 4. Get TA --->|            |

Author: ```suggestion
   (b) TA -- 2b. Supply ----------> | 4. Messaging-->|            |
```

## Malicious TA Removal

It is possible that a rogue developer distributes a malicious Untrusted 
Application and intends to get a malicious TA installed. Such a TA

> might be able to hide from malware detection the REE, or access trusted

Author: ```suggestion
might be able to escape from malware detection by the REE, or access trusted
```

> resources within the TEE (but could not access other TEEs, or access other

Reviewer: to hide from malware detection the REE, -->
to escape from malware detection by the REE
